pkg/lib/cockpit.js

```diff
@@ -1166,8 +1166,8 @@ function factory() {
             if (self.url_root)
                 pre_parts = self.url_root.split('/').map(decodeURIComponent);
 
-            if (input && input[0] !== "/") {
-                result = [].concat(self.path);
+            if (input && input[0] !== "/" && self.path !== undefined) {
+                result = [...self.path];
```

Comment:
Note that this changes behaviour: previously if self.path was undefined, it would set `result = [undefined]`, while now it skips this code path completely.

Reply:
Now it shouldn't:

result = [].concat(undefined) => result = [undefined]
result.pop()
result = []
result.concat(parts)
result = [...parts]

Reply:
ah, right!